_studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_utils.cpp

```diff
@@ -0,0 +1,488 @@
+        eMFXHWType typeHW = MFX_HW_UNKNOWN;
+        typeHW = core->GetHWType();
+
+        if (IsNeedPartialAcceleration_MPEG2(par) && platform != MFX_PLATFORM_SOFTWARE)
+        {
+            return MFX_PLATFORM_SOFTWARE;
+        }
+
+        if (platform != MFX_PLATFORM_SOFTWARE)
+        {
+#if defined (MFX_VA_LINUX)
+            if (MFX_ERR_NONE != core->IsGuidSupported(sDXVA2_ModeMPEG2_VLD, par))
+            {
+                platform = MFX_PLATFORM_SOFTWARE;
+            }
+#endif
+        }
+
+        return platform;
+    }
+
+    bool IsHWSupported(VideoCORE *core, mfxVideoParam *par)
+    {
+        if ((par->mfx.CodecProfile == MFX_PROFILE_MPEG1 && core->GetPlatformType()== MFX_PLATFORM_HARDWARE))
+            return false;
+
+#if defined (MFX_VA_LINUX)
+        if (MFX_ERR_NONE != core->IsGuidSupported(sDXVA2_ModeMPEG2_VLD, par))
+            return false;
+#endif
+
+        return true;
+    }
+
+    mfxStatus Query(VideoCORE *core, mfxVideoParam *in, mfxVideoParam *out)
+    {
+        MFX_CHECK_NULL_PTR1(out);
+        mfxStatus res = MFX_ERR_NONE;
+
+        eMFXHWType type = core->GetHWType();
+        if (!in)
+        {
+            // configurability mode
+            memset(out, 0, sizeof(*out));
+            out->mfx.CodecId = MFX_CODEC_MPEG2;
+            out->mfx.FrameInfo.FourCC = 1;
+            out->mfx.FrameInfo.Width = 1;
+            out->mfx.FrameInfo.Height = 1;
+            out->mfx.FrameInfo.CropX = 1;
+            out->mfx.FrameInfo.CropY = 1;
+            out->mfx.FrameInfo.CropW = 1;
+            out->mfx.FrameInfo.CropH = 1;
+            out->mfx.FrameInfo.AspectRatioH = 1;
+            out->mfx.FrameInfo.AspectRatioW = 1;
+
+            out->mfx.FrameInfo.FrameRateExtN = 1;
+            out->mfx.FrameInfo.FrameRateExtD = 1;
+            out->mfx.FrameInfo.PicStruct = 1;
+            out->mfx.CodecProfile = 1;
+            out->mfx.CodecLevel = 1;
+            out->mfx.ExtendedPicStruct = 1;
+            out->mfx.TimeStampCalc = 1;
+
+            out->Protected = 0;
+
+            out->mfx.FrameInfo.ChromaFormat = MFX_CHROMAFORMAT_YUV420;
+
+            if (type == MFX_HW_UNKNOWN)
+                out->IOPattern = MFX_IOPATTERN_OUT_SYSTEM_MEMORY;
+            else
+                out->IOPattern = MFX_IOPATTERN_OUT_VIDEO_MEMORY;
+
+            // DECODE's configurables
+            out->mfx.NumThread = 1;
+            out->AsyncDepth = MFX_AUTO_ASYNC_DEPTH_VALUE;
+        }
+        else
+        {
+            // checking mode
+            if (1 == in->mfx.DecodedOrder)
+                return MFX_ERR_UNSUPPORTED;
+
+            if (in->mfx.FrameInfo.FourCC != MFX_FOURCC_NV12)
+            {
+                out->mfx.FrameInfo.FourCC = 0;
+                return MFX_ERR_UNSUPPORTED;
+            }
+
+            if (in->NumExtParam != 0 && !in->ExtParam)
+                return MFX_ERR_UNSUPPORTED;
+
+            if (!in->NumExtParam && in->ExtParam)
```

Comment:
Merge 2 ifs: if( (in->NumExtParam == 0) != (in->ExtParam == nullptr) )

Reply:
ok